File: Presentation/CSCZJ.Web/src/app/viewModels/Properties/property.ts
//表格对象
export class PropertyListItem{
    id:number;
    name:string;
    propertyType:string;
    getMode?:string;
    address?:string;
    constructorArea?:number;
    landArea?:number;
    floor?:number;
    propertyId?:string;
    isAdmission:boolean;
    getedDate?:string;
    usedPeople?:string;
    fourToStation?:string;
    estateId?:string;
    constructId?:string;
    constructTime?:string;
    landId?:string;
    landTime?:string;
    currentType:string;
    useType:string;
    isMortgage:boolean;
    governmentId:number;
    governmentName:string;
    canEditDelete: boolean;   
    location:string;
}

//详细的资产对象
export class Property extends PropertyListItem {
    lon?:number;
    lat?:number;
    rents:PropertyRentModel[];
}

//新建资产对象
export class PropertyCreateModel
{    
    id:number;
    name:string;
    propertyTypeId:string;
    address:string;
    floor:number;
    fourToStation?:string;
    getedDate:string;   
    getModeId:string;
    isAdmission:string;
    registerEstate:string;
    estateId:string;
    estateTime:string;
    constructId:string;
    constructArea:number;
    constructTime:string;
    landId:string;
    landArea:number;
    landTime:string;
    landTimeStr:string;
    governmentId:string;
    governmentName:string;
    useTypeId:string;
    currentTypeId:string;
    isMortgage:string;
    description:string;
    logo:string;
    logoUrl:string;
    logoPictureId:number;
    location:string;
    extent:string;
    usedPeople:string;
    submit:boolean;
    pictures:PropertyPictureModel[];
    files:PropertyFileModel[];
    currentUse_Self:number
    currentUse_Rent:number
    currentUse_Lend:number
    currentUse_Idle:number
    owner_self:number
    owner_children:number
    price:number
    propertyNature:string
    landNature:string
    lifeTime:number 

    constructor(){
        this.name="";
        this.address="";
        this.floor=0;
        this.fourToStation="";
        this.getedDate="";   
        // this.getModeId="0";
        this.isAdmission="";
        //this.registerEstate="true";
        this.estateId="";
        this.estateTime="";
        this.constructId="";
        this.constructArea=0;
        this.constructTime="";
        this.landId="";
        this.landArea=0;
        this.landTime="";
        this.landTimeStr="";
        // this.useTypeId="0";
        // this.currentTypeId="0";
        this.isMortgage="";
        this.description="";
        this.logo="";
        this.logoUrl="";
        this.logoPictureId=0;
        this.location="";
        this.extent="";
        this.usedPeople="";
        this.submit=false;
        this.pictures=[];
        this.files=[];
        this.currentUse_Self=0;
        this.currentUse_Rent=0;
        this.currentUse_Lend=0;
        this.currentUse_Idle=0;
        this.owner_self=0;
        this.owner_children=0;
        this.price=0;
        this.propertyNature="";
        this.landNature="";
        this.lifeTime=0;        
        this.governmentName="";
    }
}

//资产图片模型
export class PropertyPictureModel
{
    pictureId:number;
    propertyId:number;
    displayOrder:number;
    title:string;
    href:string;
    isLogo:boolean;
}

//资产文件模型
export class PropertyFileModel
{
    fileId:number;
    propertyId:number;
    displayOrder:number;
    title:string;
    src:string;
    percentage:number;
    uploaded:boolean;
}

export class ApproveModel
{
    title:string;
    dSuggestion:string;
    dApproveDate:string;
    aSuggestion:string;
    aApproveDate:string;
    state:string;
    property_Id:number;
    submit:boolean;
}

export class PropertyRentModel extends ApproveModel
{
    id:number;
    name:string;
    rentTime:string;
    backTime:string;
    processDate:string;
    valid:boolean;
    priceString:string;
    priceList?:string[];
    rentArea:number;
    rentMonth:number;
    reamrk?:string;    
    rentPictures:PropertyPictureModel[];
    rentFiles:PropertyFileModel[];
}

File: Libraries/CSCZJ.Core/Domain/Properties/Property.cs
﻿using System;
using System.Collections;
using System.Collections.Generic;
using System.Data.Entity.Spatial;

namespace CSCZJ.Core.Domain.Properties
{
    public class Property : BaseEntity
    {
        public Property()
        {
        }

        private ICollection<PropertyAllot> _propertyAllots;
        private ICollection<PropertyLend> _propertyLends;
        private ICollection<PropertyRent> _propertyRents;
        private ICollection<PropertyEdit> _propertyEdits;
        private ICollection<PropertyPicture> _propertyPictures;
        private ICollection<PropertyFile> _propertyFiles;

        /// <summary>
        /// 资产名称
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// 资产类别
        /// </summary>
        public PropertyType PropertyType { get; set; }

        /// <summary>
        /// 资产所处区域
        /// </summary>
        public Region Region { get; set; }

        /// <summary>
        /// 获取方式
        /// </summary>
        public GetMode GetMode { get; set; }

        /// <summary>
        /// 地址
        /// </summary>
        public string Address { get; set; }

        /// <summary>
        /// 层数
        /// </summary>
        public int Floor { get; set; }

        /// <summary>
        /// 产权证号
        /// </summary>
        public string PropertyID { get; set; }
     
        /// <summary>
        /// 是否入账
        /// </summary>
        public bool IsAdmission { get; set; }

        /// <summary>
        /// 取得时间
        /// </summary>
        public DateTime? GetedDate { get; set; }

        /// <summary>
        /// 使用方
        /// </summary>
        public string UsedPeople { get; set; }
    
        /// <summary>
        /// 坐落位置
        /// </summary>
        public DbGeography Location { get; set; }

        public double X { get; set; }

        public double Y { get; set; }

        /// <summary>
        /// 范围，房产不需要
        /// </summary>
        public DbGeography Extent { get; set; }

        /// <summary>
        /// 描述
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// 四至情况
        /// </summary>
        public string FourToStation { get; set; }

        /// <summary>
        /// 发布状态
        /// </summary>
        public bool Published { get; set; }

        public int DisplayOrder { get; set; }

        /// <summary>
        /// 资产所有方
        /// </summary>
        public virtual GovernmentUnit Government { get; set; }

        /// <summary>
        /// 不动产证
        /// </summary>
        public string EstateId { get; set; }

        /// <summary>
        /// 不动产证发证时间
        /// </summary>
        public DateTime? EstateTime { get; set; }

        /// <summary>
        /// 房产证
        /// </summary>
        public string ConstructId { get; set; }

        /// <summary>
        /// 建筑面积
        /// </summary>
        public double ConstructArea { get; set; }

        /// <summary>
        /// 房产证发证时间
        /// </summary>
        public DateTime? ConstructTime { get; set; }

        /// <summary>
        /// 土地证号
        /// </summary>
        public string LandId { get; set;}

        /// <summary>
        /// 土地面积
        /// </summary>
        public double LandArea { get; set; }

        public DateTime? LandTime { get; set; }

        /// <summary>
        /// 使用现状
        /// </summary>
        public CurrentType CurrentType { get; set; }
        /// <summary>
        /// 用途
        /// </summary>
        public UseType UseType { get; set; }
        /// <summary>
        /// 是否抵押
        /// </summary>
        public bool IsMortgage { get; set; }

        public virtual PropertyNewCreate PropertyNewCreate { get; set; }

        public virtual ICollection<PropertyAllot> Allots
        {
            get { return _propertyAllots ?? (_propertyAllots = new List<PropertyAllot>()); }
            protected set { _propertyAllots = value; }
        }
        public virtual ICollection<PropertyLend> Lends
        {
            get { return _propertyLends ?? (_propertyLends = new List<PropertyLend>()); }
            protected set { _propertyLends = value; }
        }
        public virtual ICollection<PropertyRent> Rents
        {
            get { return _propertyRents ?? (_propertyRents = new List<PropertyRent>()); }
            protected set { _propertyRents = value; }
        }

        public virtual ICollection<PropertyEdit> Edits
        {
            get { return _propertyEdits ?? (_propertyEdits = new List<PropertyEdit>()); }
            protected set { _propertyEdits = value; }
        }

        public virtual PropertyOff PropertyOff { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public bool Locked { get; set; }

        /// <summary>
        /// 是否核销
        /// </summary>
        public bool Off { get; set; }

        /// <summary>
        /// 是否为Excel导入
        /// </summary>
        public bool FromExcelImport { get; set; }


        /// <summary>
        /// 父级资产Id，默认为0
        /// </summary>
        public int ParentPropertyId { get; set; }

        
        public virtual ICollection<PropertyPicture> Pictures
        {
            get { return _propertyPictures ?? (_propertyPictures = new List<PropertyPicture>()); }
            protected set { _propertyPictures = value; }
        }

        public virtual ICollection<PropertyFile> Files
        {
            get { return _propertyFiles ?? (_propertyFiles = new List<PropertyFile>()); }
            protected set { _propertyFiles = value; }
        }

    }
}
